core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -219,6 +219,12 @@ private[spark] class DiskBlockObjectWriter(
         truncateStream = new FileOutputStream(file, true)
         truncateStream.getChannel.truncate(committedPosition)
       } catch {
+        // ClosedByInterruptException is an excepted exception when kill task,
+        // don't log the exception stack trace to avoid confusing users.
+        // See: SPARK-28340
+        case _: ClosedByInterruptException =>
+          logError("ClosedByInterruptException occurred while reverting partial writes to file"
+            + ", it maybe caused by job cancel.")
```

Comment:
Is this second part meaningful? Maybe just `ClosedByInterruptException while reverting partial writes to  file" + file` (i.e. keep the file name)